### ISSUE TITLE
Select picture doesn't show pictures

### ISSUE THREAD

**Issue by WombatPete** [RetroShare/RetroShare#933]
When I'm sending a message to a connection (or a broadcast), there's a button to insert a picture. Oddly, the pop-up allows you to see a list of files to choose from, or a detailed list... but not PICTURES (thumbnails).

That can't be right, right?

**Comment by cavebeat** [comment]
you are talking about https://retroshare.readthedocs.io/en/latest/user-guide/interface/#broadcast
right? 

The Broadcast does not sends pictures. I'd say it sends them as a file (hashes it and sends a Download Link with the hash). 

so no, no pics in Broadcast. in Chatrooms and messaging it works. though there are some restrictions on size in chatrooms, to reduce the picture spamming.

**Comment by WombatPete** [comment]
No, I meant "broadcast" as in what you see under networks, when you're messaging a group of contacts simultaneously. It's what you're talking about in messaging. There's a button at the bottom to add a picture. It works, but the views available are all text views - the function to choose a picture displays the files only as texts, so you can't see the picture you want. Does that make sense?

**Comment by cavebeat** [comment]
i have to correct, the broadcast send pictures, as you said. my wrong, i just did not use it for years.
the link directs to "broadcast". 

i did have no problem in sending a picture with the attach button right now. 

there are a few buttons: Smiley, attach picture, attache file, settings and search. 

the attach picture selects only images, the attach file "all files (*)"

i am not able to reproduce your issue

**Comment by WombatPete** [comment]
I'm not saying it doesn't work - it does WORK!
When you open a directory on your computer using a file manager, you can generally choose what to see - a list of files; a list with details, like the date created or file type; or thumbnails. When a program has a button to select specifically a picture, you'd expect the view to be "thumbnails." Here, when you press the Attach Picture button, you get to choose a file - but the window that comes up not only doesn't DEFAULT to a view with thumbnails of your image files, it won't even LET YOU view them that way. If I don't know the pictures attached to a file name, I have to open them in another program to know what to choose. It's not fatal, but it seems like it should be easy to do it right. Does that make more sense?

**Comment by cavebeat** [comment]
sry, now i get it. ya, right, thats oddly. 

ping @PhenomRetroShare  any idea for thumbnails in the file select popup. 
similar to GIMP

**Comment by hunbernd** [comment]
It can be fixed by using the native file dialogs.
Just remove this piece of code from everywhere: QFileDialog::DontUseNativeDialog

**Comment by WombatPete** [comment]
Thanks for that. A search of the  files in the directory show a bunch of "DontUseNativeDialog"s, but no "QFileDialog"s.

**Comment by cavebeat** [comment]
as discussed in the PR above, the native picture upload dialogues with thumbnails crashed often. so it was disabled for good reasons. 

may i close

**Comment by WombatPete** [comment]
If you want. Not being in the programming mix, I'm sure there's stuff I don't understand. But it seems an odd approach.
A: Hey, this useful function, common in other programs, is causing a problem.
Two possible answers:
B: OK, let's find out why that common function is causing a problem and fix it.
or
B: Well, let's keep the problem and just eliminate the useful and common function.
Again, I'm sure there's a reason the community has apparently gone with the 2nd decision. It means that, to attach a picture, I have to go to another program that has this common function (file manager, for instance), look at the picture, remember the name (possibly difficult, esp. if it's an internet download) or rename it, come back, click on choose picture, find that file name, and choose it.
Odd.
